### ISSUE TITLE
Rendering cycleway with pedestrians

### ISSUE THREAD

**Issue by tvainika** [gravitystorm/openstreetmap-carto#1321]
Currently highway=cycleway is always rendered in blue dashed line. However, especially in continental Europe, it is very common that cycleway is shared with pedestrians (foot=yes tag) in some areas, but not in other areas. In addition, there are two types of sharing, which is tagged with segregated=yes/no.

Different types of cycle infrastructure have different quality properties. Also, in some countries different traffic rules apply (if it is obligatory for bicyclist to use that part of the road or not). Third, having different rendering highly visible makes it easier to fix tags.

Therefore, following four different cases should be rendered differently:
1. Dedicated cycleway with tags highway=cycleway + foot=no (implicit): Render non-dashed blue line.
2. Segregated cycleway and footway with tags highway=cycleway + foot=designated + segregated=yes: Render something different, for example different dashing style, which has longer lines every other time
3. Combined cycleway and footway  with tags highway=cycleway + foot=designated + segregated=no: Render something different, for example mixed blue and red in dashed line.
4. Unknown type of cycleway (highway=cycleway, none of the above): Dashed blue line as it is now.

**Comment by mkoniecz** [comment]
I have serious doubts whatever such detail is reasonable on general purpose map.

I think that it belongs rather to a special rendering displaying bicycle-related features in detail.

**Comment by kocio-pl** [comment]
I always thought that mixed foot+bocycle highways should be rendered as mixed red+blue dashed line - dead simple, of course if we don't try to rethink all the cases: than we need something like @tvainika complex proposition before we change rendering.

Speaking of that: I think segregated cycleway and footway should always be mapped as two independent highways, because they can be not really parallel in some parts (think of the tree or grass between them) and we have such a good imagery for some time (at least here in Poland), that we should start mapping and rendering them as areas. Of course, if the tagging is like you suggested, it's better to show it on the map.

**Comment by HolgerJeromin** [comment]
I don't want two independent highways for a way like this:
https://commons.wikimedia.org/wiki/File:Cycle_path_along_Kingsclere_Road_-_geograph.org.uk_-_701953.jpg

**Comment by polarbearing** [comment]
Seems @kocio-pl meant "segregated+physically separated", as she mentioned grass.

**Comment by kocio-pl** [comment]
Why not? We can easily draw different lines here, but of course areas would be even better (more sticking to the reality). I mentioned physically segregated places as an example when it is clear that it's easier to follow what we see on the ground than think how to map what we see on the traffic signs.

**Comment by RAytoun** [comment]
@tvainika <https://github.com/tvainika> The problem with suggestions
changing the depiction of features on the map is first to ensure that there
are no other features that are depicted like that. "Render non-dashed blue
line." is the same as River and Stream: or "which has longer lines every
other time" is already in use in brown as a type of track in some areas. At
present a cycleway is rendered with a blue dotted line and a footpath with
a red dotted line (not dashed lines). However alternating the colour of the
dot, as mentioned by @kocio-pl <https://github.com/kocio-pl>, to depict two
or three different uses for a way is feasible... unfortunately you are not
going to be able visually see from the rendering whether it is a cycleway
that allows pedestrians or a footway that allows bicycles. I am not sure
how easy or difficult it may be to render it.
This kind of detail is possible with the correct rendering but should only
be rendered at the higher levels of zoom and revert to less specialised
rendering for lower levels. As different levels of detail are rendered
differently at different zoom levels you should try to decide how the
features will be rendered for the level of zoom that it will appear on and
how it will be rendered at lower zoom levels.

**Comment by matkoniecz** [comment]
I agree with https://github.com/gravitystorm/openstreetmap-carto/pull/1325#issuecomment-77950410 that 

> In addition, I don't think rendering shared cycleways different from cycleway-only is really necessary.

> All cycleway-only roads I know are either of two categories: either there is a footway immediately next to it, or pedestrians are officially not allowed to use the cycleway but everyone does so anyway. In neither case, it's not really useful for pedestrians to know that they can't use the cycleway. Is this different in Finland (or anywhere else)?

(...)

> In Poland in the second case law allows pedestrians to use cycleways.

I am not completely against displaying this information - but I am not expecting it to be doable given constraints caused by displaying already so much information.

**Comment by imagico** [comment]
This is only partly related to this issue but currently highway=path with foot=designated and bicycle=designated is rendered in cycleway-blue while highway=footway with foot=designated and bicycle=designated is shown in footway-red.  So although highway=path is no more rendered distinctly it still has an influence on how other tags are interpreted.  Both from the mapper perspective and the map user perspective this is fairly strange and does not really encourage systematic use of the tags.

**Comment by matkoniecz** [comment]
@imagico There is a tradition (I am not sure whatever it is valuable to encourage it) to tag ways with diverse access as paths. For example JOSM complains about highway=footway, bicycle=designated.

**Comment by imagico** [comment]
I know - that's perfectly fine.  But the current selection of red/blue based on highway/foot/bicycle tags is inconsistent - if the intent is to make everything that is usable by bike in blue - as the blue rendering of highway=path with foot=designated and bicycle=designated implies - it should be the same for highway=footway.

Or to put it differently - in a two class system where everything is either footway or cycleway in rendering you have to decide if it is to be a map primarily for cyclists showing ways usable by bike as cycleway and the rest as footway or primarily for pedestrians showing everything usable by foot as footway and the rest as cycleway.  The current tag interpretation is a mixture:

* highway=path with bicycle=designated is shown as cycleway, even with foot=designated (implying cyclist priority)
* highway=path without foot/bicycle is shown as footway (implying pedestrian priority)
* highway=footway with bicycle=designated is shown as footway (implying pedestrian priority)

which is just confusing IMO.
